airflow/providers/amazon/aws/sensors/emr.py

```diff
@@ -587,3 +626,26 @@ def failure_message_from_response(response: dict[str, Any]) -> str | None:
                 f"with message {fail_details.get('Message')} and log file {fail_details.get('LogFile')}"
             )
         return None
+
+    def execute(self, context: Context) -> None:
+        if not self.deferrable:
+            super().execute(context=context)
+        else:
+            timeout = self.max_attempts * self.poke_interval + 60
+            self.defer(
+                timeout=timedelta(seconds=timeout),
+                trigger=EmrStepSensorTrigger(
+                    job_flow_id=self.job_flow_id,
+                    step_id=self.step_id,
+                    target_states=self.target_states,
+                    aws_conn_id=self.aws_conn_id,
+                    poke_interval=int(self.poke_interval),
+                ),
+                method_name="execute_complete",
+            )
+
+    def execute_complete(self, context, event=None):
+        if event["status"] != "success":
+            raise AirflowException(f"Error while running job: {event}")
+        else:
+            self.log.info("Job completed.")
```

Comment:
It seems to me that this else block might not be needed.
```suggestion
        if event["status"] != "success":
            raise AirflowException(f"Error while running job: {event}")

        self.log.info("Job completed.")
```